injected/unit-test/features.js

```diff
@@ -23,3 +34,83 @@ describe('Features definition', () => {
+        return schemaGenerator.createGenerator({
+            path: path.resolve(__dirname, '../../node_modules/@duckduckgo/privacy-configuration/schema/config.ts'),
+        });
+    }
+
+    function getSchema(schemaName) {
+        return createGenerator().createSchema(schemaName);
+    }
+
+    function createValidator(schemaName) {
+        const ajv = new Ajv({ allowUnionTypes: true });
+        return ajv.compile(getSchema(schemaName));
+    }
+
+    // Utility to ensure 'hash' exists on all features in the config
+    // Ideally we would not have these required in the config, it's pretty unnecessary for tests.
+    function ensureHashOnFeatures(config) {
+        if (config && typeof config === 'object' && config.features) {
+            for (const featureKey of Object.keys(config.features)) {
+                if (config.features[featureKey] && typeof config.features[featureKey] === 'object') {
+                    if (!('hash' in config.features[featureKey])) {
+                        config.features[featureKey].hash = '';
+                    }
+                }
+            }
+        }
+        return config;
+    }
+
+    const configFiles = glob
+        .sync('../integration-test/test-pages/*/config/*.json', { cwd: __dirname })
+        .map((p) => path.resolve(__dirname, p));
+
+    // Legacy allowlist: skip schema validation for these known legacy files
+    // Some of these have expected invalid configs
+    const legacyAllowlist = [
+        // Favicon configs
+        path.resolve(__dirname, '../integration-test/test-pages/favicon/config/favicon-disabled.json'),
+        path.resolve(__dirname, '../integration-test/test-pages/favicon/config/favicon-absent.json'),
+        path.resolve(__dirname, '../integration-test/test-pages/favicon/config/favicon-enabled.json'),
+        path.resolve(__dirname, '../integration-test/test-pages/favicon/config/favicon-monitor-disabled.json'),
+        // Duckplayer configs
+        path.resolve(__dirname, '../integration-test/test-pages/duckplayer/config/overlays-live.json'),
+        path.resolve(__dirname, '../integration-test/test-pages/duckplayer/config/click-interceptions-disabled.json'),
+        path.resolve(__dirname, '../integration-test/test-pages/duckplayer/config/disabled.json'),
+        path.resolve(__dirname, '../integration-test/test-pages/duckplayer/config/overlays-drawer.json'),
+        path.resolve(__dirname, '../integration-test/test-pages/duckplayer/config/overlays.json'),
+        path.resolve(__dirname, '../integration-test/test-pages/duckplayer/config/thumbnail-overlays-disabled.json'),
+        path.resolve(__dirname, '../integration-test/test-pages/duckplayer/config/video-alt-selectors.json'),
+        path.resolve(__dirname, '../integration-test/test-pages/duckplayer/config/video-overlays-disabled.json'),
+        // Message bridge configs
+        path.resolve(__dirname, '../integration-test/test-pages/message-bridge/config/message-bridge-enabled.json'),
```

Comment:
These ones specifically are actually invalid. For example if we add new bridges then this will become immediately invalid; which it doesn't matter.

For now I think this is good enough but perhaps in future we'd have a way to mark _why_ it's invalid. So the rest of the file gets schema checked still.